Jenkinsfile

```diff
@@ -118,6 +130,35 @@ pipeline {
                 }
             }
         }
+        stage('Publish to Test-pypi') {
+            when {
+                branch 'main'
+            }
+            steps {
+                script {
+                    sh "twine check dist/csvcubed*.whl"
+
+                    withCredentials([usernamePassword(credentialsId: 'testpypi-robons', usernameVariable:'TWINE_USERNAME', passwordVariable: 'TWINE_PASSWORD')]) {
+                        sh 'twine upload -r testpypi dist/csvcubed*.whl'
+                    }
```

Comment:
@robons This is self-explanatory from the name of `usernameVariable` and `passwordVariable`. But I would like to double-check. The `TWINE_USERNAME ` and `TWINE_PASSWORD` are env variables pointing to the actual username and password securely stored in jenkins right? And they are not the actual password and username?

Reply:
https://twine.readthedocs.io/en/stable/index.html#environment-variables

>   TWINE_USERNAME - the username to use for authentication to the repository.
>    TWINE_PASSWORD - the password to use for authentication to the repository.

Yes, these variables are where we store the username and password that twine needs to authenticate with pypi.